src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventProvider.cs

```diff
@@ -471,7 +471,7 @@ private unsafe void GetSessionInfo(Action<int, long> action)
 
             // Determine our session from what is in the registry.  
             string regKey = @"\Microsoft\Windows\CurrentVersion\Winevt\Publishers\{" + m_providerId + "}";
-            if (System.Runtime.InteropServices.Marshal.SizeOf(typeof(IntPtr)) == 8)
+            if (sizeof(IntPtr) == 8)
```

Comment:
In some cases I see you've used IntPtr.Size and in others sizeof(IntPtr)... what's the rationale for when one vs the other?